streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -333,11 +333,11 @@ public Subscription subscription(Set<String> topics) {
                                     } else {
                                         numPartitionsCandidate = metadata.partitionCountForTopic(sourceTopicName);
                                         if (numPartitionsCandidate == null) {
-                                            repartitionTopicMetadata.get(topicName).numPartitions = NOT_AVAILABLE;
+                                            numPartitionsCandidate = NOT_AVAILABLE;
                                         }
                                     }
 
-                                    if (numPartitionsCandidate != null && numPartitionsCandidate > numPartitions) {
+                                    if (numPartitionsCandidate == NOT_AVAILABLE || numPartitionsCandidate > numPartitions) {
```

Comment:
Will add a line.
`UNKOWN` basically means, we need to retry -- that's because information is not available for downstream operator, but will get available after upstream operators got processed -- and we don't know the order -- thus, we loop until all information was propagated.

`NOT_AVAILABLE` means missing metadata, and this is also propagated downstream (like an actual known number of partitions) to avoid an infinite loop.

Later on, we skip task creating if at least on input topic has `NOT_AVAILABLE` as numPartitions.

Makes sense?